src/jws/flattened.rs

```diff
@@ -0,0 +1,542 @@
+//! Flattened JWS signatures: see RFC 7515 section 7.2.2
+//! Flattened signatures are JSON (unlike compact signatures),
+//! and support a single signature protecting a set of headers and a
+//! payload.
+//! The RFC specifies unprotected headers as well, but this implementation
+//! doesn't support them.
+
+use super::{Header, RegisteredHeader, Secret};
+use crate::errors::{Error, ValidationError};
+use crate::jwa::SignatureAlgorithm;
+use crate::serde_custom;
+use data_encoding::BASE64URL_NOPAD;
+use serde::{de::DeserializeOwned, Deserialize, Deserializer, Serialize};
+
+// Not using CompactPart::to_bytes here, bounds are overly restrictive
+fn serialize_header<H: Serialize>(header: &Header<H>) -> Vec<u8> {
```

Comment:
We should return a `Result` since the serialization can fail and not panic with unwraps.

Reply:
Done